src/client/auth/oidc.rs

```diff
@@ -351,12 +360,52 @@ async fn do_two_step_auth(
     Ok(())
 }
 
+fn get_allowed_hosts<'a>(
+    mechanism_properties: &'a Option<Document>,
```

Comment:
Rust convention is to always accept options as concrete types, i.e. `Option<&'a Document>`.

Reply:
Ah, thanks!